src/app/clusters/joint-fabric-administrator-server/joint-fabric-administrator-server.cpp

```diff
@@ -102,31 +100,82 @@ void MatterJointFabricAdministratorPluginServerInitCallback()
     AttributeAccessInterfaceRegistry::Instance().Register(&gJointFabricAdministratorAttrAccess);
 }
 
-// TODO
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
+const FabricInfo * RetrieveCurrentFabric(CommandHandler * aCommandHandler)
+{
+    FabricIndex index = aCommandHandler->GetAccessingFabricIndex();
+    ChipLogDetail(Zcl, "JointFabric: Finding fabric with fabricIndex 0x%x", static_cast<unsigned>(index));
+    return Server::GetInstance().GetFabricTable().FindFabricWithIndex(index);
+}
+
 bool emberAfJointFabricAdministratorClusterICACCSRRequestCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::JointFabricAdministrator::Commands::ICACCSRRequest::DecodableType & commandData)
 {
     MATTER_TRACE_SCOPE("ICACCSRRequest", "JointFabricAdministrator");
+    ChipLogProgress(Zcl, "JointFabricAdministrator: Received an ICACCSRRequest command");
+
+    auto nonDefaultStatus           = Status::Success;
+    const FabricInfo * fabricInfo   = RetrieveCurrentFabric(commandObj);
+    auto & failSafeContext          = Server::GetInstance().GetFailSafeContext();
+    auto & jointFabricAdministrator = Server::GetInstance().GetJointFabricAdministrator();
+
+    uint8_t buf[Credentials::kMaxDERCertLength];
+    MutableByteSpan icacCsr(buf, Credentials::kMaxDERCertLength);
+    Commands::ICACCSRResponse::Type response;
+
+    // If current fabric is not available, command was invoked over PASE which is not legal
+    VerifyOrExit(fabricInfo != nullptr, nonDefaultStatus = Status::InvalidCommand);
```

Comment:
The command can be invoked over PASE (after AddNOC) and you would still get a non-null fabricInfo here.

If you want to check for CASE, you should be checking for CASE, not other things which don't exactly match the "is this over CASE?" state.

See `GeneralCommissioningGlobalInstance::HandleArmFailSafe` where it does the `authMode == Access::AuthMode::kCase` check for the right way to test this.